lib/Dialect/TorchConversion/Transforms/BackendTypeConversionPasses.cpp

```diff
@@ -115,6 +115,15 @@ static void setupFinalization(ConversionTarget &target,
   setupFinalization<OpTy2, OpTys...>(target, patterns, typeConverter);
 }
 
+static void stripTorchAttrs(func::FuncOp func) {
+  SmallVector<NamedAttribute> newAttrs;
+  for (auto attr : func->getDialectAttrs()) {
```

Comment:
I'm not very familiar with how torch-mlir uses attributes. This could also look through `getAllArgAttrs()` and `getAllResultAttrs()`.

Reply:
It is relatively expensive to mutate these attributes. Let's make it only pay that cost if there are attributes to change. Can you carry a `bool modified` outside the loop to note if you dropped any and then only setDialectAttrs if modified?

Reply:
Added a modified check and just a comment about how "arg" and "result" attrs could also be stripped if needed.